integration/maptest/map_test.go

```diff
@@ -15,409 +15,54 @@
 package maptest
 
 import (
-	"bytes"
 	"context"
-	"crypto"
-	"encoding/hex"
+	"flag"
 	"fmt"
+	"log"
+	"os"
 	"testing"
 
-	"github.com/google/trillian"
-	"github.com/google/trillian/crypto/keys/der"
-	"github.com/google/trillian/examples/ct/ctmapper/ctmapperpb"
-	"github.com/google/trillian/merkle"
-	"github.com/google/trillian/merkle/hashers"
-	"github.com/google/trillian/testonly"
 	"github.com/google/trillian/testonly/integration"
 
-	tcrypto "github.com/google/trillian/crypto"
-	stestonly "github.com/google/trillian/storage/testonly"
-
 	_ "github.com/google/trillian/merkle/coniks"
 	_ "github.com/google/trillian/merkle/maphasher"
 )
 
-var h2b = testonly.MustHexDecode
-
-// createBatchLeaves produces n unique map leaves.
-func createBatchLeaves(batch, n int) []*trillian.MapLeaf {
-	leaves := make([]*trillian.MapLeaf, 0, n)
-	for i := 0; i < n; i++ {
-		leaves = append(leaves, &trillian.MapLeaf{
-			Index:     testonly.TransparentHash(fmt.Sprintf("batch-%d-key-%d", batch, i)),
-			LeafValue: []byte(fmt.Sprintf("batch-%d-value-%d", batch, i)),
-		})
-	}
-	return leaves
-}
-
-func isEmptyMap(ctx context.Context, env *integration.MapEnv, tree *trillian.Tree) error {
-	r, err := env.MapClient.GetSignedMapRoot(ctx, &trillian.GetSignedMapRootRequest{
-		MapId: tree.TreeId,
-	})
-	if err != nil {
-		return fmt.Errorf("failed to get empty map head: %v", err)
-	}
-
-	if got, want := r.GetMapRoot().GetMapRevision(), int64(0); got != want {
-		return fmt.Errorf("got SMR with revision %d, want %d", got, want)
-	}
-	return nil
-}
-
-func verifyGetMapLeavesResponse(getResp *trillian.GetMapLeavesResponse, indexes [][]byte,
-	wantRevision int64, pubKey crypto.PublicKey, hasher hashers.MapHasher, treeID int64) error {
-	if got, want := len(getResp.MapLeafInclusion), len(indexes); got != want {
-		return fmt.Errorf("got %d values, want %d", got, want)
-	}
-	if got, want := getResp.GetMapRoot().GetMapRevision(), wantRevision; got != want {
-		return fmt.Errorf("got SMR with revision %d, want %d", got, want)
-	}
+var (
+	server = flag.String("map_rpc_server", "", "Server address:port")
+	env    *integration.MapEnv
+)
 
-	// SignedMapRoot contains its own signature. To verify, we need to create a local
-	// copy of the object and return the object to the state it was in when signed
-	// by removing the signature from the object.
-	smr := *getResp.GetMapRoot()
-	smr.Signature = nil // Remove the signature from the object to be verified.
-	if err := tcrypto.VerifyObject(pubKey, smr, getResp.GetMapRoot().GetSignature()); err != nil {
-		return fmt.Errorf("VerifyObject(SMR): %v", err)
-	}
-	rootHash := getResp.GetMapRoot().GetRootHash()
-	for _, incl := range getResp.MapLeafInclusion {
-		leaf := incl.GetLeaf().GetLeafValue()
-		index := incl.GetLeaf().GetIndex()
-		leafHash := incl.GetLeaf().GetLeafHash()
-		proof := incl.GetInclusion()
+func TestMain(m *testing.M) {
```

Comment:
Could we keep `TestMain` in `main_test.go`?   Would simplify the diff.

Reply:
Also, do we still need a TestMain?

Reply:
We do still need TestMain, as it is the only opportunity to install the pre-test hooks for the environment.

Reply:
Yes, we could keep TestMain in main_test.go